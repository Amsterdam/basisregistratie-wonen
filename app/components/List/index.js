import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Ul = styled.ul`
  padding-left: 0;

  &.depth-1 {
    padding: 0 0 30px;

    ul {
      padding: 0;
    }
  }
  list-style: none;

  &.depth-2 {
    & > li {
      padding-left: 0;

      & > ul {
        padding-bottom: 30px;

        & > li.has-list {
          padding-left: 0 !important;
        }
      }

      & :before {
        content: none;
      }
    }
  }

  &:not(:first-child) {
    border-bottom: 4px solid #767676;
  }

  li {
    padding-left: 23px;
    position: relative;

    &:before {
      content: '';
      position: absolute;
      left: 0;
      width: 8px;
      height: 8px;
      margin-top: 7px;
      background-color: #767676;
    }
  }
`;

const List = ({ children, ...rest }) => <Ul {...rest}>{children}</Ul>;

List.propTypes = {
  children: PropTypes.node.isRequired,
};

export default List;
