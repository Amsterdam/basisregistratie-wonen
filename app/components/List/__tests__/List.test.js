import React from 'react';
import { render, cleanup } from 'react-testing-library';
import 'jest-styled-components';

import List from '..';

describe('List', () => {
  afterEach(cleanup);

  it('should match the snapshot', () => {
    const { container } = render(
      <List>
        <li>Item #1</li>
        <li>Item #2</li>
        <li>Item #3</li>
      </List>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
