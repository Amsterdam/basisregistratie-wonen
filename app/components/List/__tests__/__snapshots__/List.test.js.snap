// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should match the snapshot 1`] = `
.c0 {
  padding: 0;
  list-style: none;
  padding-bottom: 30px;
}

.c0:not(:first-child) {
  border-bottom: 4px solid #767676;
}

.c0 li:last-of-type ul {
  padding-bottom: 0;
}

.c0 *:last-of-type > *:last-of-type ul:last-of-type {
  border: 0;
}

.c0 li {
  padding-left: 23px;
  position: relative;
}

.c0 li:before {
  content: '';
  position: absolute;
  left: 0;
  width: 8px;
  height: 8px;
  margin-top: 7px;
  background-color: #767676;
}

.c0 ul.is-nested {
  padding-bottom: 0;
}

.c0 ul.is-nested ul:not(:last-of-type) {
  padding-bottom: 1em;
}

.c0 ul.is-nested > li {
  padding-left: 0;
}

.c0 ul.is-nested > li:before {
  content: none;
}

<ul
  class="c0"
>
  <li>
    Item #1
  </li>
  <li>
    Item #2
  </li>
  <li>
    Item #3
  </li>
</ul>
`;
